Add keyboard events to CheckBox component for better keyboard a11y

crocodele opened the pull request:
CheckBox component can be navigated to using Tab and field value can be toggled using space bar (same as native checkbox inputs).

No style changes – focus outline etc. – included.

antlau commented:
Would it be possible to set the focus to the native input-element (that is currently non-visible) and let the browser flip styles and states with natively triggers that include the spacebar? Yes, it will probably require some style changes, but it would be native behavior without explicit event listeners.

crocodele commented:
Focus on native input element doesn't seem to be possible with `visibility: hidden` or `display: none`, but `opacity: 0` does the trick. I removed the event listeners and added outline styles on focus.
